[Cron] Fix handling bogus last/next times + restructure code

TD-er opened the pull request:
The plugin was assuming the values stored in UserVar were correct.
After a cold boot, or when the NTP time (or via GPS) was not yet set, these values for last time or next time could be just about any value.
This means it was possible the Cron task would never fire an event if the next time was computed far in the future.

Also restructured the code a bit, to make it more readable.

TD-er committed:
[Cron] Fix handling bogus last/next times + restructure code

The plugin was assuming the values stored in UserVar were correct.
After a cold boot, or when the NTP time (or via GPS) was not yet set, these values for last time or next time could be just about any value.
This means it was possible the Cron task would never fire an event if the next time was computed far in the future.

Also restructured the code a bit, to make it more readable.

TD-er committed:
[Cron] Fix mistake of generating incorrect event task name.

uzi18 commented on src/_P081_Cron.ino:
```diff
@@ -176,168 +220,156 @@ boolean Plugin_081(byte function, struct EventStruct *event, String& string)
+      break;
     }
 
     case PLUGIN_TEN_PER_SECOND:
```
@TD-er why not remove this part?

TD-er replied:
What part?

The `break;` is needed as it is part of the case statement.
And the success is set after an event is added to the event queue.

The success is not used as return value from the  `PLUGIN_ONCE_A_SECOND` call, but I thought it would make it more clear for a reader to show when we achieved what we were aiming for.


Or do you mean the "empty" cases?

uzi18 replied:
@TD-er section PLUGIN_TEN_PER_SECOND

TD-er committed:
[Cron] Call systmTimePresent() function to check if we have any time set

We do need to have some idea of time for this plugin.

TD-er committed:
[Cron] Only update last/next times if cron elapsed

When also updating the last/next times as the time is updated, a cron interval may be skipped while it was just about to be executed.
